libs/tracing/unix/Tracing.ml

```diff
@@ -51,24 +51,83 @@ module Otel = Opentelemetry
  * TODO we'll probably need instructions for some system of tags?
  *)
 
+(*****************************************************************************)
+(* Types *)
+(*****************************************************************************)
+
+type asdf_span = Int64.t [@@deriving show]
+
+type analysis_flags = {
+  secrets_validators : bool;
+  allow_all_origins : bool;
+  historical_scan : bool;
+  deep_intra_file : bool;
+  deep_inter_file : bool;
+}
+[@@deriving show]
+
+type top_level_data = { version : string; analysis_flags : analysis_flags }
+[@@deriving show]
+
 (*****************************************************************************)
 (* Constants *)
 (*****************************************************************************)
 
 let default_endpoint = "https://telemetry.dev2.semgrep.dev"
 let endpoint_env_var = "SEMGREP_OTEL_ENDPOINT"
 
+(*****************************************************************************)
+(* Helpers *)
+(*****************************************************************************)
+
+let no_analysis_features () =
+  {
+    secrets_validators = false;
+    historical_scan = false;
+    allow_all_origins = false;
+    deep_intra_file = false;
+    deep_inter_file = false;
+  }
+
+(* Set the descriptor for allowed origins. This is not simply
+   a boolean because we will likely include new origins in the
+   future *)
+let allowed_origins allow_all_origins =
+  if allow_all_origins then "all_origins" else "pro_rules_only"
+
+(* Poor man's Git repo detection. Running git repo detection again
+   seems wasteful, but checking two env vars is pretty cheap.
+
+   TODO the more we port of semgrep scan and semgrep ci, the more
+   of this information will already be in OCaml *)
+let repo_name () =
+  match Sys.getenv_opt "SEMGREP_REPO_DISPLAY_NAME" with
+  | Some name -> name
+  | None -> (
+      match Sys.getenv_opt "SEMGREP_REPO_NAME" with
+      | Some name -> name
+      | None -> "<local run>")
+
+(* In case we don't have a repo name, report the base folder where
+   semgrep was run. We report only the base name to avoid leaking
+   user information they may not have expected us to include. *)
+let current_working_folder () = Filename.basename (Sys.getcwd ())
```

Comment:
Do we need a realpath or anything here? I recall there being some discussion around something possibly related.

Reply:
Discussed in office; determined that this should be fine